[EDU-4077] feat: add usage guides for console kit (#748)
aziontech/docs: src/content/docs/pt-br/pages/guias/devtools/integrar-com-api/index.mdx

---

> title: Como integrar o Console Azion e a API da Azion usando o Console Kit

Reviewer: ```suggestion
title: Como integrar o Console da Azion e a API da Azion usando o Console Kit
```

description: O Console Kit da Azion permite que você crie uma interface Azion personalizada para atender às necessidades do seu negócio. Neste guia, você aprenderá a mostrar mais informações da API da Azion no Console.
meta_tags: edge, console, development, interface, vue, primevue
namespace: documentation_devtools_console_kit_api
menu_namespace: devtoolsMenu
permalink: /documentacao/produtos/guias/integrar-com-api-console-kit/
---

O **Console Kit** da Azion permite que você crie uma interface Azion personalizada para atender às necessidades do seu negócio. Neste guia, você aprenderá a mostrar mais informações da API da Azion no Console.

:::caution[warning]

> Para seguir este guia, você deve primeiro [configurar seu projeto Console Kit](/en/documentation/devtools/console-kit/#setting-up-a-project) em um repositório bifurcado na sua conta e inicializar o projeto usando o CLI da Azion para executá-lo localmente. Então, você pode trabalhar na alteração da versão bifurcada do Console da Azion e, uma vez concluído, entregar o projeto para a edge usando o CLI.

Reviewer: ```suggestion
Para seguir este guia, você deve primeiro [configurar seu projeto do Console Kit](/en/documentation/devtools/console-kit/#setting-up-a-project) em um repositório bifurcado (forked) na sua conta e inicializar o projeto usando a CLI da Azion para executá-lo localmente. Então, você pode trabalhar na alteração da versão bifurcada do Console da Azion e, uma vez concluído, entregar o projeto no edge usando a CLI.
```

:::

Para integrar *APIs externas* com a interface do Console da Azion, você pode criar um novo serviço Axios API em `src/services/axios`, adaptar cada serviço conforme necessário e seguir as etapas abaixo.

---

> ## Obter dados da API da Azion e mostrar na interface do Console

Reviewer: ```suggestion
## Obtenha dados da API da Azion e mostre na interface do Console
```

Neste exemplo, você modificará um serviço no Console para mostrar dois campos adicionais da API da Azion na tabela de configurações de cache da aplicação edge.

> O objetivo é mostrar os valores TTL para cache do navegador e cache da edge, informados pelas propriedades `browser_cache_settings_maximum_ttl` e `cdn_cache_settings_maximum_ttl` na [API de Cache Settings](https://api.azion.com/#fc4c8368-a64d-4743-9064-58e03786edf4). Para fazer isso:

Reviewer: ```suggestion
O objetivo é mostrar os valores TTL para cache do navegador e cache do edge, informados pelas propriedades `browser_cache_settings_maximum_ttl` e `cdn_cache_settings_maximum_ttl` na [API de Cache Settings](https://api.azion.com/#fc4c8368-a64d-4743-9064-58e03786edf4). Para fazer isso:
```

> 1. Abra o projeto Console Kit no seu IDE.

Reviewer: ```suggestion
1. Abra o projeto do Console Kit no seu IDE.
```

2. Execute `azion dev` para inicializar um servidor de desenvolvimento local.
3. Na pasta `src/services`, localize o `list-cache-settings-service.js` e adicione as novas propriedades ao mapa:

```javascript title="src/services/edge-application-cache-settings-services/list-cache-settings-service.js"
const adapt = (httpResponse) => {
  const parseHttpResponse = httpResponse.body.results.map((cacheSettings) => ({
    id: cacheSettings.id.toString(),
    name: cacheSettings.name,
    browserCache: cacheSettings.browser_cache_settings,
    cdnCache: cacheSettings.cdn_cache_settings,
    // inicializa as seguintes referências baseado nos campos da API
    browserCacheTtl: cacheSettings.browser_cache_settings_maximum_ttl,
    edgeCacheTtl: cacheSettings.cdn_cache_settings_maximum_ttl
  }))

  return {
    body: parseHttpResponse,
    statusCode: httpResponse.statusCode
  }
}
```

> 5. Agora localize a **Cache Settings** list view e modifique-a da seguinte forma:

Reviewer: ```suggestion
4. Agora localize a list view **Cache Settings** e modifique-a da seguinte forma:
```

```javascript title="src/views/EdgeApplicationsCacheSettings/ListView.vue"
  const getColumns = computed(() => {
    return [
      {
        field: 'name',
        header: 'Origin Name'
      },
      {
        field: 'browserCache',
        header: 'Browser Cache'
      },
      // cria a coluna browser cache TTL
      {
        field: 'browserCacheTtl',
        header: 'Browser Cache TTL'
      },
      {
        field: 'cdnCache',
        header: 'Edge Cache'
      },
      // cria a coluna edge cache TTL
      {
        field: 'edgeCacheTtl',
        header: 'Edge Cache TTL'
      }
    ]
  })
```

> 6. No navegador, acesse o endereço localhost e navegue até a página **Edge Applications**.
> 7. Crie uma aplicação edge ou selecione uma existente.
> 8. Navegue até a aba **Cache Settings**. Agora você deve ver as novas colunas aparecerem na UI.
> 9. Depois de terminar, execute `azion deploy` para lançar as alterações para a edge.

Reviewer: ```suggestion
5. No navegador, acesse o endereço localhost e navegue até a página **Edge Applications**.
6. Crie uma edge application ou selecione uma existente.
7. Navegue até a aba **Cache Settings**. Agora você deve ver as novas colunas aparecerem na interface.
8. Depois de terminar, execute `azion deploy` para lançar as alterações no edge.
```
